include/util/guidance/name_announcements.hpp

```diff
@@ -129,16 +131,34 @@ inline bool requiresNameAnnounced(const std::string &from_name,
 
     const auto pronunciation_changes = from_pronunciation != to_pronunciation;
 
-    return !obvious_change || needs_announce || pronunciation_changes;
+    // when exiting onto ramps, we need to be careful about exit numbers. These can often be only
+    // assigned to the first part of the ramp
+    //
+    //  a . . b . c . . d
+    //         ` e . . f
+    //
+    // could assign the exit number to `be` when exiting `abcd` instead of the full ramp.
+    //
+    // Issuing a new-name instruction here would result in the turn assuming the short segment to be
+    // irrelevant and remove the exit number in a collapse scenario. We don't want to issue any
+    // instruction from be-ef, since we only loose the exit number. So we want to make sure that we
+    // don't just loose an exit number, when exits change
+    const auto exits_change = from_exits != to_exits;
+    const auto looses_exit = (names_are_equal && !from_exits.empty() && to_exits.empty());
```

Comment:
I'm unclear on what the overall effect here is supposed to be, please clarify on which these are true:

1/ We emit a new-name instruction at `be->ef`
2/ We don't emit a new-name instruciton at `be->ef`
3/ We just ensure these steps won't be collapsed
4/ We just ensure the exit number will be copied to the new step